x-pack/plugins/triggers_actions_ui/server/data/routes/indices.ts

```diff
@@ -53,28 +53,31 @@ export function createIndicesRoute(logger: Logger, router: IRouter, baseRoute: s
     }
 
     let aliases: string[] = [];
-    try {
-      aliases = await getAliasesFromPattern(esClient, pattern);
-    } catch (err) {
-      logger.warn(`route ${path} error getting aliases from pattern "${pattern}": ${err.message}`);
-    }
-
     let indices: string[] = [];
+    let dataStreams: string[] = [];
+
     try {
-      indices = await getIndicesFromPattern(esClient, pattern);
+      const results = await Promise.allSettled([
```

Comment:
`Promise.allSettled()` does **not** throw if one of the promises is rejected, so the `catch` part would only get thrown for an unexpected runtime error, I think.  Specifically, the `catch` won't be executed if one of the `get*FromPattern()` calls rejects.

At least that's my understanding - doesn't seem like it's [doc'd that well at MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) - you'd think they'd note that as a significant difference from `Promise.all()`.  

I tested it manually with this single line from a node REPL, and it printed "did not catch it".

`try { await Promise.allSettled([Promise.reject(new Error('wops'))]); console.log('did not catch it') } catch (err) { console.log('caught it') }`

We'll need to either throw in the `.map()` call if we detect a rejection, or use a different method like `Promise.all()`.

It seems like we should have some jest tests for this, which would have caught this ... but this plugin is lacking jest tests for all the routes - can we add one for this route, and then open a ticket to add tests for the other routes?

Reply:
We spoke offline and decided to revert the parallel optimization. Updated in this [commit](https://github.com/elastic/kibana/pull/137584/commits/a7f797973eeb82365b7feb98de6c003edada63b8)

Reply:
Might as well add some comments on why we reverted here - or you could add to the new issue to open up to optimize these calls and reference this issue in the new issue.